src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/RememberMeFactory.php

```diff
@@ -31,7 +31,68 @@ class RememberMeFactory implements SecurityFactoryInterface
 
     public function create(ContainerBuilder $container, $id, $config, $userProvider, $defaultEntryPoint)
     {
-        // authentication provider
+        $authProviderId = $this->configureRememberMeAuthenticationProvider($container, $id, $config);
+
+        $this->addRememberMeServicesToLogoutListener($container, $id, $config);
+
+        $this->configureRememberMeServices($container, $id, $config);
+
+        $listenerId = $this->configureRememberMeAuthenticationListener($container, $id, $config);
+
+        return array($authProviderId, $listenerId, $defaultEntryPoint);
+    }
+
+    public function getPosition()
+    {
+        return 'remember_me';
+    }
+
+    public function getKey()
+    {
+        return 'remember-me';
+    }
+
+    public function addConfiguration(NodeDefinition $node)
+    {
+        $node->fixXmlConfig('user_provider');
+        $builder = $node->children();
+
+        $builder
+            ->scalarNode('key')->isRequired()->cannotBeEmpty()->end()
+            ->scalarNode('token_provider')->end()
+            ->arrayNode('user_providers')
+                ->beforeNormalization()
+                    ->ifString()->then(function ($v) { return array($v); })
+                ->end()
+                ->prototype('scalar')->end()
+            ->end()
+        ;
+
+        foreach ($this->options as $name => $value) {
+            if (is_bool($value)) {
+                $builder->booleanNode($name)->defaultValue($value);
+            } else {
+                $builder->scalarNode($name)->defaultValue($value);
+            }
+        }
+    }
+
+    private function getRememberMeServicesId($id, array $config)
```

Comment:
I'm not sure if I should assume that `$config` is always array (here and in other methods). If not, I'll remove the typehints.